File: JavaScript/8js 3/uniTotal.js
/*
|--------------------------------------------------------------------------
| ASCII Total
|--------------------------------------------------------------------------
|
*/

function uniTotal(string) {
   return string == ""
      ? 0
      : string
           .split("")
           .map((i) => i.charCodeAt(0))
           .reduce((a, b) => a + b);
}


function uniTotal(string) {
   return [...string].reduce((acc, char) => acc + char.charCodeAt(0), 0);
}


function uniTotal(string) {
   let count = 0;
   for (let i = 0; i < string.length; i++) {
      count += string.charCodeAt(i);
   }
   return count;
}


console.log(uniTotal(""), 0);
console.log(uniTotal("e"), 101);
console.log(uniTotal("aaa"), 291);
console.log(uniTotal("Mary Had A Little Lamb"), 1873);

File: Python/8py 3/uni_total.py
"""
|--------------------------------------------------------------------------
| ASCII Total
|--------------------------------------------------------------------------
|
"""

def uni_total(s):
   return sum([ord(i) for i in s])


def uni_total(string):
   res = 0
   for i in string:
      res += ord(i)
   return res


def uni_total(string):
   return sum(map(ord, string))


print(uni_total(""), 0)
print(uni_total("e"), 101)
print(uni_total("aaa"), 291)
print(uni_total("Mary Had A Little Lamb"), 1873)